### PR TITLE
Add QCoroImageProvider

### PR DESCRIPTION
**Pull request by JBBgameich**
Sorry for starting another pull request in parallel :)
I already had this code in a project of mine and think it might be useful to upstream.

### CODE REVIEW

**Review by danvratil — CHANGES_REQUESTED** [review]
Nice, thanks for upstreaming this! Looks OK, just two small nitpicks.

Also please update `.github/workflows/generate-matrix.py` and add `qtdeclarative` to "archives" field for both Qt5 and Qt6 maps (at the beginning of the file), that should make the pipelines pass.

If you are interested in writing a unit-test for the class as well, I'd appreciate it (you can do it in another PR), but it's not requirement (I'll write it later).

File: docs/reference/quick/imageprovider.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<!--
+SPDX-FileCopyrightText: 2022 Jonah Brüchert <jbb@kaidan.im>
+SPDX-License-Identifier: GFDL-1.3-or-later
+-->
+
+# ImageProvider
+
+{{ doctable("Core", "QCoroImageProvider") }}
```
[the remark this answers is not in the document]
Comment (JBBgameich) [reply docs/reference/quick/imageprovider.md]: Oops, must have missed that while copy-pasting.

**Comment by danvratil** [comment]
The CI is failing on Linux because I forgot that the Docker images used in the CI don't contain the qtdeclarative module. I've rebuild the images, so the next run should succeed.

**Review by danvratil — APPROVED** [review]

**Comment by danvratil** [comment]
Could you please rebase on current `main`? Afterwards I'll merge it :)

**Comment by danvratil** [comment]
Thank you!

**Pull request merged**